tests/components/element-card/element-card.spec.tsx

```diff
@@ -0,0 +1,109 @@
+import { fireEvent, render, screen } from "@testing-library/react";
+import "jest-canvas-mock";
+import { ElementCard } from "src/components";
+import { EElementType, TElement } from "src/context-providers";
+import { ELEMENT_BUTTON_LABELS } from "src/data/elements-data";
+import { TestHelper, mockBuilderState } from "src/util/test-helper";
+
+describe("element-card", () => {
+    afterEach(() => {
+        jest.restoreAllMocks();
+        jest.resetAllMocks();
+    });
+
+    describe("onClick", () => {
+        it("should be in a focused state and show the edit element panel when clicked on", () => {
+            renderComponent(
+                { element: MOCK_ELEMENT, onClick: mockOnClick },
+                {}
+            );
+            fireEvent.click(getElementCard() as HTMLElement);
+            expect(mockOnClick).toHaveBeenCalled();
+        });
+    });
+
+    describe("rendering of duplicate & delete button", () => {
+        it("should not show the duplicate and delete buttons when in normal state", () => {
+            renderComponent({ element: MOCK_ELEMENT });
+            expect(getDuplicateButton(true)).not.toBeInTheDocument();
+            expect(getDeleteButton(true)).not.toBeInTheDocument();
+        });
+
+        it("should show the duplicate and delete buttons when in focused state", () => {
+            renderComponent(
+                { element: MOCK_ELEMENT },
+                {
+                    builderContext: {
+                        ...mockBuilderState,
+                        focusedElement: { element: MOCK_ELEMENT },
+                    },
+                }
+            );
+            expect(getDuplicateButton()).toBeInTheDocument();
+            expect(getDeleteButton()).toBeInTheDocument();
+        });
+        it("should disable the duplicate button when there is already a duplicated element card", () => {
+            renderComponent(
+                { element: MOCK_ELEMENT },
+                {
+                    builderContext: {
+                        ...mockBuilderState,
+                        focusedElement: {
+                            element: MOCK_ELEMENT,
+                            isDirty: true,
+                            isValid: false,
+                        },
+                    },
+                }
+            );
+            expect(getDuplicateButton()).toBeInTheDocument();
+            expect(getDuplicateButton()).toBeDisabled();
+        });
+    });
+});
+
```

Comment:
Add these test cases too
- should not render acton buttons if the focused element is not valid (set `isDirty` to false)
- delete button click event (check that the `deleteElement` in `useBuilder` is called with the current element's internal id)

Reply:
You'll need to mock `useBuilder` for the second test. Here is an example how
```tsx
const mockDeleteElement = jest.fn();

jest.mock("src/context-providers/builder/hook.tsx", () => {
	const actual = jest.requireActual("src/context-providers/builder/hook.tsx");
	return {
		useBuilder: () => ({
			...actual.useBuilder(), // we don't want to mock `focusedElement`
			deleteElement: mockDeleteElement
		}),
	};
});
```